### PR TITLE
dounload csv

### PR DESCRIPTION
**Pull request by varvarashcherbyna7**

### CODE REVIEW

File: src/App.tsx [hunk 1]
```diff
@@ -30,13 +32,16 @@ function App() {
 
   const [fullScreenMode, setFullScreenMode]= useState<boolean>(false);
 
+  // const location = useLocation();
+  // const splitLocation = location.pathname.split("/");
+  // const api_key = splitLocation[splitLocation.length - 2];
+  // const [client, setClient] = useState<Client>(ClientDefault);
```
Comment (bezryma) [inline src/App.tsx]: do we really need this commented code?)

File: src/api/reportApi.ts [hunk 2]
```diff
@@ -32,4 +32,27 @@ export const reportApi = {
...
+        start_time: data.startDate,
+        end_time: data.endDate
+    }
+
+    try {
+      const response = await instance()
+      .post('api/client/report_new_clients', dataForBack)
+      console.log('response report new clients ', response);
+
+      // console.log('response report new clients received successfully ', response.headers["content-length"]);
```
Comment (bezryma) [inline src/api/reportApi.ts:51]: Can we remove this commented log and make code cleaner?)

File: src/components/ClientInfo/Name/Name.tsx [hunk 3]
```diff
@@ -44,6 +44,18 @@ export default function Name(): ReactElement {
     getClient()
   }, []);
 
+  useEffect(() => {
+    setActiveBtnRogueMode(activeBtnRogueMode);
+    // if (activeBtnRogueMode === "on") {
+    //   history.push('/queue');
```
Comment (bezryma) [inline src/components/ClientInfo/Name/Name.tsx:50]: do we need this comment?)

File: src/components/Queue/Queue.tsx [hunk 4]
```diff
@@ -53,6 +55,16 @@ export default function Queue(): ReactElement {
     getClientsForQueue();
   }, []);
 
+  useEffect(() => {
+    setActiveBtnRogueMode(activeBtnRogueMode);
+    // if (activeBtnRogueMode === "on") {
+    //   history.push('/queue');
```
Comment (bezryma) [inline src/components/Queue/Queue.tsx]: do we need this comment?)

File: src/components/ClientInfo/Name/NameOn.tsx [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+                </div>
+                <div className="alphaContainer_dashboard">
+                  <IntakeAlpha/>
+                </div>
+              </div>
+            </div>
+        </div>
+    </div>
+  )
+}
```
Comment (bezryma) [inline src/components/ClientInfo/Name/NameOn.tsx:177]: so many commented code in this file))

File: src/components/Reports/Reports.tsx [hunk 6]
```diff
@@ -137,38 +138,55 @@ export default function Reports(): ReactElement {
...
-    const dateStart = new Date(data.startDate.toISOString().replace(/GMT.*$/,'GMT+0000')).toISOString();
-    const dStart = dateStart.replace('T', ' ').replace(".000Z", " ").split(" ");
-    const fullDateStart = dStart[0].split("-");
-    const fullTime = dStart[1];
-    const startDateToBack = `${fullDateStart[1]}/${fullDateStart[2]}/${fullDateStart[0]}, ${fullTime}`
-    console.log("startDateToBack ", startDateToBack);
+    const fullStartDate = data.startDate.toISOString().replace("T", " ").replace(".", " ").split(" ");
+    const dStart = fullStartDate[0].split("-");
+    // console.log(dStart);
+    // console.log(fullStartDate[1]);
```
Comment (bezryma) [inline src/components/Reports/Reports.tsx]: pls, remove this comments)

File: src/components/Reports/Reports.tsx [hunk 7]
```diff
@@ -137,38 +138,55 @@ export default function Reports(): ReactElement {
...
+
+    const fullTime = fullStartDate[1];
+    const startDateToBack = `${dStart[1]}/${dStart[2]}/${dStart[0]}, ${fullTime}`
+    // console.log("startDateToBack ", startDateToBack);
```
Comment (bezryma) [inline src/components/Reports/Reports.tsx]: pls, remove this comments)

File: src/components/Reports/Reports.tsx [hunk 8]
```diff
@@ -137,38 +138,55 @@ export default function Reports(): ReactElement {
...
-    const endStart = new Date(data.endDate.toISOString().replace(/GMT.*$/,'GMT+0000')).toISOString();
-    const dEnd = endStart.replace('T', ' ').replace(".000Z", " ").split(" ");
-    const fullDateEnd = dEnd[0].split("-");
-    const fullTimeEnd = dEnd[1];
-    const endDateToBack = `${fullDateEnd[1]}/${fullDateEnd[2]}/${fullDateEnd[0]}, ${fullTimeEnd}`
-    console.log("startDateToBack ", endDateToBack);
+    const fullEndDate = data.endDate.toISOString().replace("T", " ").replace(".", " ").split(" ");
+    const dEnd = fullEndDate[0].split("-");
+    // console.log(dEnd);
+    // console.log(fullEndDate[1]);
```
Comment (bezryma) [inline src/components/Reports/Reports.tsx]: pls, remove this comments)

File: src/components/Reports/Reports.tsx [hunk 9]
```diff
@@ -137,38 +138,55 @@ export default function Reports(): ReactElement {
...
 
+    const fullTimeEnd = fullEndDate[1];
+    const endDateToBack = `${dEnd[1]}/${dEnd[2]}/${dEnd[0]}, ${fullTimeEnd}`
+    // console.log("startDateToBack ", endDateToBack);
```
Comment (bezryma) [inline src/components/Reports/Reports.tsx]: pls, remove this comment)

File: src/components/Reports/Reports.tsx [hunk 10]
```diff
@@ -137,38 +138,55 @@ export default function Reports(): ReactElement {
...
 
     const typeString = Object.values(data.type)[0].toLowerCase()
-    console.log("type", typeString);
+    // console.log("type", typeString);
```
Comment (bezryma) [inline src/components/Reports/Reports.tsx]: pls, remove this comment)

File: src/components/Reports/Reports.tsx [hunk 11]
```diff
@@ -137,38 +138,55 @@ export default function Reports(): ReactElement {
...
 
-  console.log("fileVisits", fileVisits);
+    if (dataToBack.type === 'new_clients') {
+      reportApi.filterDateToReportNewClients(dataToBack);
+      getFileWithNewClients();
+      // if (fileVisits) {
+      //   getFileWithNewClients();
+      // }
+        // alert(`There are no new clients during this period ${startDateToBack} - ${endDateToBack}`);
+        // console.log("There are no new clients during this period")
```
Comment (bezryma) [inline src/components/Reports/Reports.tsx]: do we need this comment?

File: src/components/Reports/Reports.tsx [hunk 12]
```diff
@@ -137,38 +138,55 @@ export default function Reports(): ReactElement {
...
 
-
     if (dataToBack.type === 'visits') {
       reportApi.filterDateToReportVisit(dataToBack);
       getFileWithVisits();
+      // if (fileVisits) {
+      //   getFileWithVisits();
+      // }
+        // alert(`No visits during this period ${startDateToBack} - ${endDateToBack}`);
+        // console.log("No visits for this date")
```
Comment (bezryma) [inline src/components/Reports/Reports.tsx]: do we need this comment?

**Review by bezryma — COMMENTED** [review]
New code looks great, but there are many use(full/less) comments!

File: src/components/ClientInfo/Name/NameOn.tsx (hunk 5 above)
Comment (varvarashcherbyna7) [reply src/components/ClientInfo/Name/NameOn.tsx:177]: Thank you, Dana for your comments, I really need some comments yet) I'm deleting all the others now))

**Pull request merged**